pkg/cmd/secvulnMaven.go

```diff
@@ -5,33 +5,299 @@
 package cmd
 
 import (
+	"encoding/xml"
 	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
 
 	"github.com/spf13/cobra"
 )
 
 var (
-    secvulnMavenCmd = &cobra.Command{
+	secvulnMavenCmd = &cobra.Command{
 		Use:   "maven",
 		Short: "Generate psuedo maven project for security vulnerability scanning",
 		Long:  "Generate psuedo maven project for security vulnerability scanning",
 		Run:   secvulnMavenExecute,
 	}
 
-	secvulnMavenParentDir     string
-	secvulnMavenPomUrls    *[]string
+	secvulnMavenParentDir string
+	secvulnMavenPomUrls   *[]string
+	secvulnMavenPomRepos  *[]string
+
+	completedProjects []string
+	toDoProjects      []string
 )
 
 func init() {
 	secvulnMavenCmd.PersistentFlags().StringVar(&secvulnMavenParentDir, "parent", "", "Parent project directory")
 	secvulnMavenPomUrls = secvulnMavenCmd.PersistentFlags().StringArray("pom", nil, "Component Pom URLs")
+	secvulnMavenPomRepos = secvulnMavenCmd.PersistentFlags().StringArray("repo", nil, "Repos to look for Poms")
 
 	secvulnMavenCmd.MarkPersistentFlagRequired("parent")
 	secvulnMavenCmd.MarkPersistentFlagRequired("pom")
+	secvulnMavenCmd.MarkPersistentFlagRequired("repo")
 
 	secvulnCmd.AddCommand(secvulnMavenCmd)
 }
 
 func secvulnMavenExecute(cmd *cobra.Command, args []string) {
 	fmt.Printf("Galasa Build - Security Vulnerability Maven - version %v\n", rootCmd.Version)
+
+	for _, pom := range *secvulnMavenPomUrls {
+		startScanningPipeline(pom)
+	}
+
+	updateParent()
+	fmt.Println("Security scanning project pom.xml updated")
+
+}
+
+func startScanningPipeline(mainPomUrl string) {
+
+	fmt.Println("Starting the scanning pipeline at " + mainPomUrl)
+
+	mainPom, err := readPomFromUrl(mainPomUrl)
+	if err != nil {
+		fmt.Println("Unable to find the main pom at " + mainPomUrl)
+		panic(err)
+	}
+
+	createPseudoMavenProject(mainPom)
+	fmt.Println("Pseudo maven project created for: " + mainPom.ArtifactId)
+
+	// Repeat the process for all dependencies with groupId dev.galasa
+	fmt.Println("Pseudo maven projects created for dependency chain of " + mainPom.ArtifactId + ":")
+	for len(toDoProjects) > 0 {
+
+		var index = len(toDoProjects) - 1
+		var artifactName = toDoProjects[index]
+
+		if artifactName == "com.jcraft.jsch" {
+			toDoProjects = removeItemFromArrayByIndex(toDoProjects, index)
+			continue
+		}
+
+		if bool := checkIfCompleted(artifactName); bool == true {
```

Comment:
Love it! But you might wanna just have the boolean output from the `checkIfCompleted` method used as the if boolean rather than setting it to another variable to be only used in this if